Switch to sass-embedded and update eslint and stylelint

NeOMakinG opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 
Please take the time to edit the "Answers" rows below with the necessary information.
Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | We need to update from node-sass to sass-embedded to fix some compatibilities issues and improve build performances, at the same time, I updated eslint dependencies
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Should build as usual with node 14 and 16 and styles should be fine on FO
| Possible impacts? | Dev tool and styling

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

Commit:
Switch to sass-embedded and make it compatible with npm 8

Commit:
Revert eslint

Commit:
Update CI

Commit:
Update stylelint

matks commented:
@NeOMakinG Should we mark this as a BC break for next release, and mention in a Changelog (like "BE CAREFUL! We changed THIS STUFF") ?

> We need to update from node-sass to sass-embedded to fix some compatibilities issues and improve build performances, at the same time, I updated eslint dependencies

NeOMakinG commented:
> @NeOMakinG Should we mark this as a BC break for next release, and mention in a Changelog (like "BE CAREFUL! We changed THIS STUFF") ?
> 
> > We need to update from node-sass to sass-embedded to fix some compatibilities issues and improve build performances, at the same time, I updated eslint dependencies

This change shouldn't impact anything on user side, it shouldn't break the front-end part, but it might require you to run a newer node version to build the theme, I guess it shouldn't be a BC at all

HanaRebaiQA reviewed (commented):
Hello @NeOMakinG 

I have checked your PR. Here is my feedback.

- Check with `nodejs v16` and `npm v8` : An error is generated when run the **npm install && npm run build** under `classic/ _dev/ ` folder

![image](https://user-images.githubusercontent.com/92912932/199215953-ffe5b466-553b-44f9-bfd5-0f87c8609f6e.png)

When checking the FO : NOK

![image](https://user-images.githubusercontent.com/92912932/199216003-43ac93f1-e295-4e43-a89b-188e34461de8.png)

- Check with `nodejs v14` and `npm v6` : An error related to npm is generated when run the **npm install && npm run build** under `classic/ _dev/ ` folder

![image](https://user-images.githubusercontent.com/92912932/199217040-84b464ae-372c-4ab2-b782-69057e617394.png)

Could you please check ?

Thank You!

kpodemski commented:
Hi @NeOMakinG 

Do you think it's doable to do that for the next version of Classic?

NeOMakinG commented:
> Hi @NeOMakinG
> 
> Do you think it's doable to do that for the next version of Classic?

why not?

leemyongpakvn commented:
@NeOMakinG @HanaRebaiQA For checking with `nodejs v16` and `npm v8`, it looks like a problem with missing sass-embedded-linux-x64, you can fix it by `npm i sass-embedded-linux-x64`.
I see no problem with a fresh install of PS 8.0.1, classic-theme 2.0.6 base (manual copy PR changes :) in  `nodejs v16` and `npm v8` environment.

NeOMakinG commented:
@leemyongpakvn I suggest not doing this `npm i sass-embedded-linux-x64` it should work without it, I suspect some cache issues

leemyongpakvn commented:
@NeOMakinG It is just a quick fix. We can remove the whole node_modules directory then `npm install` instead.

leemyongpakvn commented on .github/workflows/lint.yml:
```diff
@@ -10,7 +10,7 @@ jobs:
       - name: Setup Node
         uses: actions/setup-node@v1
```
actions/setup-node@v3

leemyongpakvn commented on .github/workflows/lint.yml:
```diff
@@ -23,7 +23,7 @@ jobs:
       - name: Setup Node
         uses: actions/setup-node@v1
```
actions/setup-node@v3

leemyongpakvn reviewed (commented):
It's right time to update actions/setup-node to v3.
